reconcile/automated_actions/config/integration.py

```diff
@@ -0,0 +1,286 @@
+from reconcile.utils.runtime.integration import (
+    PydanticRunParams,
+    QontractReconcileIntegration,
+)
+from reconcile.utils.semver_helper import make_semver
+
+QONTRACT_INTEGRATION = "automated-actions-config"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+
+class AutomatedActionsConfigIntegrationParams(PydanticRunParams):
+    thread_pool_size: int
+    use_jump_host: bool
+    internal: bool | None = None
+    configmap_name: str = "automated-actions-config"
+
+
+class AutomatedActionsRole(BaseModel):
+    user: str
+    role: str
+
+
+class AutomatedActionsPolicy(BaseModel):
+    sub: str
+    obj: str
+    params: dict[str, str] = {}
+
+
+class AutomatedActionsConfigIntegration(
+    QontractReconcileIntegration[AutomatedActionsConfigIntegrationParams]
+):
+    """Manage LDAP groups based on App-Interface roles."""
+
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def get_early_exit_desired_state(
+        self, query_func: Callable | None = None
+    ) -> dict[str, Any]:
+        """Return the desired state for early exit."""
+        if not query_func:
+            query_func = gql.get_api().query
+        return {
+            "automated_actions_instances": [
+                c.dict() for c in self.get_automated_actions_instances(query_func)
+            ]
+        }
+
+    def get_automated_actions_instances(
+        self, query_func: Callable
+    ) -> Generator[AutomatedActionsInstanceV1, None, None]:
+        """Return all automated actions."""
+        data = instance_query(query_func, variables={})
+        for instance in data.automated_actions_instances_v1 or []:
+            if instance.deployment.delete:
+                continue
+            instance.permissions = list(
+                self._filter_permissions(instance.permissions or [])
+            )
+            yield instance
+
+    def _is_enabled(self, argument: AutomatedActionArgumentV1) -> bool:
+        """Check if the integration is enabled for the given argument namespace."""
+        if isinstance(argument, AutomatedActionArgumentOpenshiftV1):
+            return (
+                integration_is_enabled("automated-actions", argument.namespace.cluster)
+                and not argument.namespace.delete
+            )
+        return True
+
+    def _filter_permissions(
+        self, permissions: Iterable[PermissionAutomatedActionsV1]
+    ) -> Generator[PermissionAutomatedActionsV1, None, None]:
+        """Filter out expired roles and arguments (cluster.namespace) with disabled integrations."""
+        for permission in permissions:
+            # automated actions disabled for the cluster?
+            permission.arguments = [
+                arg for arg in permission.arguments or [] if self._is_enabled(arg)
+            ]
+            # remove expired roles
+            permission.roles = expiration.filter(permission.roles)
+            if not permission.roles:
+                continue
```

Comment:
when `roles` is empty, `permission` is skipped, but empty `arguments` still yield `permission`, why different?

Reply:
A permission without a role association has no user relations; therefore, it can be skipped. 
Not all actions have or need arguments; therefore, those permissions must still be considered.